File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Utils/Environment.swift
import Foundation

/// Centralized environment configuration for production/development switching
/// Named AppEnvironment to avoid conflict with SwiftUI's Environment property wrapper
public enum AppEnvironment {
    case development
    case staging
    case production

    /// Current environment based on build configuration
    public static var current: AppEnvironment {
        #if DEBUG
            return .development
        #else
            return .production
        #endif
    }

    /// Base URL for API requests
    public var apiBaseURL: String {
        switch self {
        case .development:
            // Use production API for development (change to localhost when running local server)
            return "https://api.lfvs.in/api"
        case .staging:
            return "https://staging.lfvs.in/api"
        case .production:
            return "https://api.lfvs.in/api"
        }
    }

    /// Host for image URLs
    public var imageHost: String {
        switch self {
        case .development:
            // Use production host for development (change to localhost when running local server)
            return "https://api.lfvs.in"
        case .staging:
            return "https://staging.lfvs.in"
        case .production:
            return "https://api.lfvs.in"
        }
    }

    /// Whether debug logging is enabled
    public var isDebugLoggingEnabled: Bool {
        switch self {
        case .development, .staging:
            return true
        case .production:
            return false
        }
    }
}

File: apps/Applications/android/app/src/main/java/com/ecommerceearn/app/utils/Environment.kt
package com.ecommerceearn.app.utils

enum class AppEnvironment {
    Development,
    Staging,
    Production;

    companion object {
        val current: AppEnvironment
            // Default to Development for debug builds; override with BuildConfig when enabled
            get() = Development
    }

    val apiBaseURL: String
        get() = when (this) {
            Development -> "https://api.lfvs.in/api"
            Staging -> "https://staging.lfvs.in/api"
            Production -> "https://api.lfvs.in/api"
        }

    val imageHost: String
        get() = when (this) {
            Development -> "https://api.lfvs.in"
            Staging -> "https://staging.lfvs.in"
            Production -> "https://api.lfvs.in"
        }

    val isDebugLoggingEnabled: Boolean
        get() = when (this) {
            Development, Staging -> true
            Production -> false
        }

    val sseEndpoint: String
        get() = when (this) {
            Development -> "https://api.lfvs.in/api/sse/events"
            Staging -> "https://staging.lfvs.in/api/sse/events"
            Production -> "https://api.lfvs.in/api/sse/events"
        }
}

object Environment {
    val current: AppEnvironment = AppEnvironment.current
}
